### PR TITLE
Kaa-818 Smart persistence of endpoint's state

### PR DESCRIPTION
**Pull request by forGGe**

### CODE REVIEW

File: client/client-multi/client-c/src/kaa/kaa_context.h [hunk 1]
```diff
@@ -95,7 +93,7 @@ typedef struct kaa_status_holder_t                  kaa_status_holder_t;
  * independently to perform API calls to specific subsystems.
  */
 typedef struct kaa_context_s {
-    kaa_status_holder_t         *status;                 /**< See @link kaa_status.h @endlink. */
+    kaa_status_t                *status;                 /**< See @link kaa_status.h @endlink. */
```
Comment (ghost) [inline client/client-multi/client-c/src/kaa/kaa_context.h]: kaa_status_holder_t is used to hide kaa_status_t  context from a user of Kaa library.
Why did you remove this holder?

File: client/client-multi/client-c/src/kaa/kaa_notification_manager.c [hunk 2]
```diff
@@ -1058,6 +1058,8 @@ static kaa_error_t update_sequence_number(kaa_notification_manager_t *self, uint
             KAA_FREE(state);
             return KAA_ERR_NOMEM;
         }
+
+        self->status->has_update = true;
     } else {
         state = (kaa_topic_state_t *)kaa_list_get_data(it);
         if (sqn_number > state->sqn_number) {
```
Comment (ghost) [inline client/client-multi/client-c/src/kaa/kaa_notification_manager.c:1064]: This flag should be raised there as well (in case of sequence number is updated).

**Comment by ohamov** [comment]
LGTM

File: client/client-multi/client-c/src/kaa/kaa_status.c [hunk 3]
```diff
@@ -139,11 +167,14 @@ kaa_error_t kaa_status_save(kaa_status_t *self)
 {
     KAA_RETURN_IF_NIL(self, KAA_ERR_BADPARAM);
 
+    if (!self->has_update)
```
Comment (ghost) [inline client/client-multi/client-c/src/kaa/kaa_status.c:170]: Need to set this flag to false in the end of the function.

**Comment by ghost** [comment]
LGTM
